doc/guide/r_quickstart.hpp

```diff
@@ -0,0 +1,191 @@
+correct <- sum(output$predictions == prepdata$test_labels)
+cat(correct, "out of", length(prepdata$test_labels), "test points correct",
+    correct / length(prepdata$test_labels) * 100.0, "%\n")
+@endcode
+
+We can see that we achieve reasonably good accuracy on the test dataset (80%+);
+if we use the full @c covertype.csv.gz, the accuracy should increase
+significantly (but training will take longer).
+
+It's easy to modify the code above to do more complex things, or to use
+different mlpack learners, or to interface with other machine learning toolkits.
+
+@section r_quickstart_whatelse What else does mlpack implement?
+
+The example above has only shown a little bit of the functionality of mlpack.
+Lots of other commands are available with different functionality.  A full list
+of each of these commands and full documentation can be found on the following
+page:
+
+ - <a href="https://www.mlpack.org/doc/mlpack-git/r_documentation.html">r documentation</a>
+
+For more information on what mlpack does, see https://www.mlpack.org/.
+Next, let's go through another example for providing movie recommendations with
+mlpack.
+
+@section r_quickstart_movierecs Using mlpack for movie recommendations
+
+In this example, we'll train a collaborative filtering model using mlpack's
+<tt><a href="https://www.mlpack.org/doc/mlpack-git/r_documentation.html#cf">cf()</a></tt> method.  We'll train this on the MovieLens dataset from
+https://grouplens.org/datasets/movielens/, and then we'll use the model that we
+train to give recommendations.
+
+You can copy-paste this code directly into R to run it.
+
+@code{.R}
+if(!requireNamespace("data.table", quietly = TRUE)) { install.packages("data.table") }
+suppressMessages({
+    library("mlpack")
+    library("data.table")
+})
+
+# First, load the MovieLens dataset.  This is taken from files.grouplens.org/
+# but reposted on mlpack.org as unpacked and slightly preprocessed data.
+ratings <- fread("http://www.mlpack.org/datasets/ml-20m/ratings-only.csv.gz")
+movies <- fread("http://www.mlpack.org/datasets/ml-20m/movies.csv.gz")
+
+# Hold out 10% of the dataset into a test set so we can evaluate performance.
+predata <- preprocess_split(input = ratings,
+                            test_ratio = 0.1,
+                            verbose = TRUE)
+
+# Train the model.  Change the rank to increase/decrease the complexity of the
+# model.
+output <- cf(training = predata$training,
+             test = predata$test,
+             rank = 10,
+             verbose = TRUE,
+             max_iteration=2,
+             algorithm = "RegSVD")
+cf_model <- output$output_model
+
+# Now query the 5 top movies for user 1.
+output <- cf(input_model = cf_model,
+             query = matrix(1),
+             recommendations = 10,
+             verbose = TRUE)
+
+# Get the names of the movies for user 1.
+cat("Recommendations for user 1:\n")
+for (i in 1:10) {
+  cat("  ", i, ":", as.character(movies[output$output[i], 3]), "\n")
```

Comment:
Good point @coatless!  @Yashwants19 I took a look through the code in this repository and in the `RcppMLPACK` repository and I think that we aren't correctly handling the difference between R's 1-indexing and C++'s 0-indexing.  I believe that we will need to modify, e.g., `SetParamUCol()` and `SetParamURow()` to subtract 1 from the values when converting to C++, and add 1 to the values when converting back to R.  That is because when we represent labels in C++, we expect them to take values between 0 and (the number of classes - 1), whereas in R it seems like we would expect a user to provide data that has classes between 1 and the number of classes.  Let me know if I overlooked something and the code already does that. :+1:

Reply:
Thank you for noticing the issue, I have tried to resolve this issue as suggested in the last commit.